### ISSUE TITLE
[pwm/runtime] CLK_DIV creating pulses too long to practically simulate

### ISSUE THREAD

**Issue by jason-hoddinett** [lowRISC/opentitan#12308]
> "All outputs are pulsed at a common clock frequency, which can be controlled relative to the PWM core clock by a 27-bit clock divider. The overall pulse frequency depends on this clock-divider and the phase-resolution."

> "The period between output pulses is 2^(DC_RESN+1)*(CLK_DIV+1) times longer than the input clock period."

This can lead to pulses too long to practically simulate. Eg, worst case scenario: 

At full resolution DC_RESN+1 = 16, maximum clock divider of CLK_DIV+1 = 134,217,727, core_clk_i = 24MHz
So, at 24MHz, the period is (24*10^6)^-1 =  42ns
The period between output pulses therefore 16 x 134,217,727 x 42 x (10^-9) = approx 90 seconds.

Is this interpretation of the HWIP spec correct? If so, how can this be verified?

**Comment by martin-lueker** [comment]
@weicaiyang @tjaychen: This does seem like a reasonable concern.  Though I'm eager to change the RTL, who do talk to get recommendations for practical ranges for coverage?

**Comment by weicaiyang** [comment]
> This does seem like a reasonable concern. Though I'm NOT eager to change the RTL, who do talk to get recommendations for practical ranges for coverage?

it's not practical to test large values on clock-divider and phase-resolution,  as it may take forever to finish a test.
I would probably constrain DC_RESN*CLK_DIV less than a fixed value, so that most of seeds can be finished within 30mins
it's also good to constrain each of these CSR values to have higher change to get the minimum value, in order cover this extreme case.
```
  DC_RESN dict {[0:5] :/ 5, [6:100] :/ 2, [101:1000] :/1};
  CLK_DIV dict {[0:5] :/ 5, [6:100] :/ 2, [101:1000] :/1};
```

**Comment by tjaychen** [comment]
for the larger values, just to make sure that nothing is wrong in the system and that it can behave correctly, do we want to consider using the TB to force a larger value in as a directed test?

**Comment by weicaiyang** [comment]
> for the larger values, just to make sure that nothing is wrong in the system and that it can behave correctly, do we want to consider using the TB to force a larger value in as a directed test?

good idea, this will improve the coverage. We need to inform scb/monitor with the same forced value.

**Comment by dharanendrak** [comment]
This issue has been addressed with the pwm_stress_all testcase
